### ISSUE TITLE
sealed-secrets generates invalid X.509 certificates by default

### ISSUE THREAD

**Issue by roy-work** [bitnami-labs/sealed-secrets#398]
The `sealed-secrets` controller will generate a new key & certificate when it is started (I think); the certificate it generates, by default, will not be valid, due to the `Issuer` and `Subject` fields being empty sequences.

`sealed-secrets` generates a certificate roughly here: https://github.com/bitnami-labs/sealed-secrets/blob/0eda19ec5420b3794484658790e9427eddd91918/pkg/crypto/keys.go#L39-L49

`cn`, in this context, is controlled by a flag given to the controller here: https://github.com/bitnami-labs/sealed-secrets/blob/0eda19ec5420b3794484658790e9427eddd91918/cmd/controller/main.go#L44

There, you can see the default value is the empty string.

One might think this would result in an Issuer of (in pseudo-syntax) `[{CN=""}]`, however, [the golang standard library](https://golang.org/src/crypto/x509/pkix/pkix.go) ignores the empty string:

```
	if len(n.CommonName) > 0 {
		ret = n.appendRDNs(ret, []string{n.CommonName}, oidCommonName)
	}
```
And so no `Issuer`/`Subject` gets set.

As for whether an empty `Issuer` is legal, [RFC 5280 §4.1.2.4](https://tools.ietf.org/html/rfc5280#section-4.1.2.4) states,

> The issuer field MUST contain a non-empty distinguished name (DN).

(The standard is more roundabout about `Subject`, but I believe it must also be non-empty, since it must match `Issuer` on certs, and `Issuer` cannot be empty.)

Something as simple as a default of `"Sealed Secrets Controller"` in that flag would suffice.

----------

(I am building a tool to monitor all of my org's certificates, and the parsing library I am using raised an error on these certificates. My reading of the standard is that the parser is correct, these are not valid. Note, however, `openssl` at least seems to tolerate them. Regardless, I think having a descriptive default is better.)

**Comment by armingerten** [comment]
This issue seems to be related to https://github.com/bitnami-labs/sealed-secrets/issues/355 . 

Is there any plan to either having set a default DN as `Issuer` or providing it during startup (e.g. analogous to the `myCN` flag as a `myIssuer` flag)?

WDYT?

**Comment by mkmik** [comment]
> Something as simple as a default of "Sealed Secrets Controller" in that flag would suffice.

PR welcome. It doesn't seem a blocking issue so I won't likely investing some time personally; but glad to merge a simple and clear PR fixes this, provided backwards compatibility is maintained.

**Comment by mkmik** [comment]
closed by #564
